bindings/fomac/fomac.cpp

```diff
@@ -26,23 +27,52 @@ namespace py = pybind11;
 using namespace py::literals;
 
 PYBIND11_MODULE(MQT_CORE_MODULE_NAME, m, py::mod_gil_not_used()) {
+  auto session = py::class_<fomac::Session>(m, "Session");
+
+  const fomac::SessionConfig defaultConfig;
+  session.def(
+      py::init(
+          [](const std::optional<std::string>& token,
+             const std::optional<std::string>& authFile,
+             const std::optional<std::string>& authUrl,
+             const std::optional<std::string>& username,
+             const std::optional<std::string>& password,
+             const std::optional<std::string>& projectId) -> fomac::Session {
+            fomac::SessionConfig config;
+            config.token = token;
+            config.authFile = authFile;
+            config.authUrl = authUrl;
+            config.username = username;
+            config.password = password;
+            config.projectId = projectId;
+            return fomac::Session{config};
+          }),
+      "token"_a = defaultConfig.token, "auth_file"_a = defaultConfig.authFile,
+      "auth_url"_a = defaultConfig.authUrl,
+      "username"_a = defaultConfig.username,
+      "password"_a = defaultConfig.password,
+      "project_id"_a = defaultConfig.projectId);
+
```

Comment:
Wouldn't it be simpler here to remove `defaultConfig` and simply use `std::nullopt` for the default values?

Reply:
I based this off @ystade's example he posted here: https://github.com/munich-quantum-toolkit/core/pull/1355#discussion_r2588318278